### ISSUE TITLE
[D3D9] Jittery Lava in Star Wars: The Force Unleashed Ultimate Sith Edition

### ISSUE THREAD

**Issue by Sterophonick** [doitsujin/dxvk#3204]
### Software information
Star Wars: The Force Unleashed Ultimate Sith Edition (AppID `32430`)
Settings: 1080P High

### System information
- Steam Deck
- Driver: Mesa 22.2.0 (git-17e5312102)
- Wine version: Proton `experimental-7.0-20230113`
- DXVK version: [v2.0-d020f4451a](https://github.com/ValveSoftware/dxvk/tree/d020f4451a3f6fb98ec36590f767f44e85dd51dc)

### Apitrace file(s)
[SWTFU Launcher.trace (544.8 MB)](https://anonfiles.com/O97aj8Tfy3/SWTFU_Launcher.trace_zip)

### Log files
[SWTFU Launcher_d3d9.log](https://github.com/doitsujin/dxvk/files/10472985/SWTFU.Launcher_d3d9.log)
[SWTFU Launcher_dxgi.log](https://github.com/doitsujin/dxvk/files/10472986/SWTFU.Launcher_dxgi.log)

### Video
https://user-images.githubusercontent.com/34801996/213885984-b44eb74b-852b-420e-8c2a-ad6f40790837.mp4

### Correct behavior
Observable at https://youtu.be/SBt-7_oGG14?t=118

### Save File
Place [SWTFU.BIN](https://github.com/doitsujin/dxvk/files/10472996/SWTFU.BIN.zip) in `%USERPROFILE%\Documents\Aspyr\Star Wars The Force Unleashed\`

**Comment by K0bin** [comment]
Thanks for the good bug report. I'll try to take a look soon.

**Comment by Blisto91** [comment]
This originally regressed in dxvk 1.9.4 by the commit that set `d3d9.floatEmulation` to Strict by default on RADV. Setting back to true back then fixed it.
It then also started appearing even with it set to true in 1.10.2 & 2.0 by the commit that set `d3d9.invariantPosition` to true by default.
So on current master
```
d3d9.floatEmulation = True
d3d9.invariantPosition = False
```
makes the flicker go away

**Comment by Sterophonick** [comment]
> This originally regressed in dxvk 1.9.4 by the commit that set `d3d9.floatEmulation` to Strict by default on RADV. Setting back to true back then fixed it. It then also started appearing even with it set to true in 1.10.2 & 2.0 by the commit that set `d3d9.invariantPosition` to true by default. So on current master
> 
> ```
> d3d9.floatEmulation = True
> d3d9.invariantPosition = False
> ```
> 
> makes the flicker go away

This seems to have not fixed the issue, and DXVK is detecting my config.
```
info:  Game: SWTFU Launcher.exe
info:  DXVK: v2.0
info:  Found config file: /run/media/deck/SteamDeckSD/steamapps/common/Star Wars The Force Unleashed/dxvk.conf
info:  Effective configuration:
info:    d3d9.invariantPosition = False
info:    d3d9.floatEmulation = True
info:  Built-in extension providers:
info:    Win32 WSI
```

**Comment by Blisto91** [comment]
Thanks for checking that out ingame. I only tried the trace with my R9 380x.

**Comment by Blisto91** [comment]
Bisecting the actual jittery liquids with the trace still leads me to the same commits as with the Z-Fighting suggesting it's also floating point stuff. Probably good idea that i also check ingame

**Comment by K0bin** [comment]
This renders correctly on Nvidia, even with `d3d9.floatEmulation = Strict`.

**Comment by Blisto91** [comment]
Appears to be fixed in mesa 23.1. Just waiting for confirmation from OP.

**Comment by K0bin** [comment]
User confirmed on Discord that the issue is fixed.
